src/pyfncall.jl

```diff
@@ -0,0 +1,168 @@
+
+# pyarg_tuples[i] is a pointer to a python tuple of length i-1
+# const pyarg_tuples = Vector{PyPtr}(32)
+const pyarg_tuples = PyPtr[]
+
+"""
+Low-level version of `pycall!(ret, o, ...; kwargs...)`
+Sets `ret.o` to the result of the call, and returns `ret::PyObject`
+"""
+function _pycall!(ret::PyObject, o::Union{PyObject,PyPtr}, args, kwargs)
+    if isempty(kwargs)
+        kw = C_NULL
+    else
+        kw = PyObject(Dict{AbstractString, Any}([Pair(string(k), v) for (k, v) in kwargs]))
+    end
+    _pycall!(ret, o, args, length(args), kw)
+end
+
+"""
+Low-level version of `pycall!(ret, o, ...)` for when `kw` is already in python
+friendly format but you don't have the python tuple to hold the arguments
+(`pyargsptr`). Sets `ret.o` to the result of the call, and returns `ret::PyObject`.
+"""
+function _pycall!(ret::PyObject, o::Union{PyObject,PyPtr}, args,
+  nargs::Int=length(args), kw::Union{Ptr{Nothing}, PyObject}=C_NULL)
```

Comment:
Shouldn't this be a `PyPtr` rather than `Ptr{Nothing}`?

Reply:
No, that's for the `C_NULL` when there aren't any kwargs. Which reminded me to add some tests for the kwargs case.

Reply:
`Ptr{Nothing}` is an implementation detail; it's best to use `Ptr{Cvoid}` (Base in 0.7, available from Compat in 0.6)

Reply:
ta